packages/connect/src/types/settings.ts

```diff
@@ -35,4 +36,5 @@ export interface ConnectSettings {
     env: 'node' | 'web' | 'webextension' | 'electron' | 'react-native';
     timestamp: number;
     proxy?: Proxy;
+    logger?: (message: LogMessage) => void;
```

Comment:
nitpick: maybe `logWriter` to make the variable name similar to what is used insede debug.ts?